File: 0322-coin-change/0322-coin-change.java
class Solution {
    public int coinChange(int[] coins, int amount) {
        int[] dp=new int[amount+1];
        dp[0]=0;
         for(int i=1;i<amount+1;i++){
            dp[i]=Integer.MAX_VALUE;
            for(int c:coins){
                if(c<=i && dp[i-c]!=Integer.MAX_VALUE){
                dp[i]=Math.min(dp[i],1+dp[i-c]);
                }
            }
         }
         if(dp[amount]==Integer.MAX_VALUE){return -1;}
         return dp[amount];
    }
}

File: 0322-coin-change/0322-coin-change.py
class Solution(object):
    def coinChange(self, coins, amount):
        """
        :type coins: List[int]
        :type amount: int
        :rtype: int
        """
        dp=[float('inf')]*(amount+1)
        dp[0]=0
        if amount==0:
            return 0

        min_coin=float('inf')
        for i in range(0,amount+1):
            
            for coin in coins:
                if coin<=i and dp[i-coin]!=float('inf'):
                    dp[i]=min(dp[i],1+dp[i-coin])
        if dp[amount]==float('inf'):
            return -1
        return dp[amount]

        